src/app/containers/Article/index.jsx

```diff
@@ -1,75 +1,54 @@
-import React, { Fragment } from 'react';
-import { bool, string, shape } from 'prop-types';
-import Helmet from 'react-helmet';
-import HeaderContainer from '../Header';
+import React from 'react';
+import { string, shape } from 'prop-types';
+import { compose } from 'recompose';
 import FooterContainer from '../Footer';
 import articlePropTypes from '../../models/propTypes/article';
-import { ServiceContextProvider } from '../../contexts/ServiceContext';
-import { RequestContextProvider } from '../../contexts/RequestContext';
-import GlobalStyle from '../../lib/globalStyles';
 import ArticleMain from '../ArticleMain';
-import ErrorMain from '../ErrorMain';
-import nodeLogger from '../../helpers/logger.node';
-import ConsentBanner from '../ConsentBanner';
+import PageWrapper from '../PageWrapper';
 
-const logger = nodeLogger(__filename);
-
-/*
-  [1] This handles async data fetching, and a 'loading state', which we should look to handle more intelligently.
-*/
-const ArticleContainer = ({ loading, error, data, bbcOrigin }) => {
-  if (loading) return 'Loading...'; /* [1] */
-  if (error) {
-    logger.error(error);
-    return 'Something went wrong :(';
-  }
+import WithError from '../PageWrapper/withError';
+import WithLoading from '../PageWrapper/withLoading';
 
+const ArticleContainer = ({ data, bbcOrigin }) => {
   if (data) {
-    const { isAmp, data: articleData, service, status } = data;
-
-    return (
-      <Fragment>
-        <GlobalStyle />
-        <ServiceContextProvider service={service}>
-          <RequestContextProvider
-            platform={isAmp ? 'amp' : 'canonical'}
-            bbcOrigin={bbcOrigin}
-          >
-            <Helmet>
-              <link
-                rel="manifest"
-                href={`/${service}/articles/manifest.json`}
-              />
-            </Helmet>
-            <ConsentBanner />
-            <HeaderContainer />
-            {status === 200 ? (
-              <ArticleMain articleData={articleData} />
-            ) : (
-              <ErrorMain status={status} />
-            )}
-            <FooterContainer />
-          </RequestContextProvider>
-        </ServiceContextProvider>
-      </Fragment>
-    );
+    const { data: articleData, isAmp, service, status } = data;
+    try {
+      return (
+        <PageWrapper isAmp={isAmp} bbcOrigin={bbcOrigin} service={service}>
+          {status === 200 && articleData ? (
+            <ArticleMain articleData={articleData} />
+          ) : (
+            <ErrorMain status={status} />
+          )}
+          <FooterContainer />
+        </PageWrapper>
+      );
+    } catch {
+      return (
+        <PageWrapper isAmp={isAmp} bbcOrigin={bbcOrigin} service={service}>
```

Comment:
Seeing as this doesnt require data, could this be the default return instead of `null`? Also, if theres an error in say, `PageWrapper`, the try catch would catch it, but then the `catch` would just cause the error again

Reply:
Good points, so Haven't looked at this part of the code too much yet, was just focussing on the original loading and error state. But we could have a WithData HOC that handles this same try catch logic here?

Reply:
We can remove the try catch here, and use Error Boundaries in App to toggle the `error` state, therefor showing the error page

Reply:
Unfortunately from the looks of things react error boundaries do not yet work for SSR applications.

The lifecycle methods used for reacts error boundaries e.g. `componentDidCatch` are not yet supported when using `renderToString` or `renderToStaticMarkup`

Reply:
😢